pkg/apis/apps/v1alpha1/update_priority.go

```diff
@@ -0,0 +1,48 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// UpdatePriorityStrategy is the strategy to define priority for pods update.
+// Only one of orderPriority and weightPriority can be set.
+type UpdatePriorityStrategy struct {
+	// Order priority terms, pods will be sorted by the value of orderedKey.
+	// For example:
+	// ```
+	// orderPriority:
+	// - orderedKey: key1
+	// - orderedKey: key2
+	// ```
+	// First, all pods which have key1 in labels will be sorted by the value of key1.
+	// Then, the left pods which have no key1 but have key2 in labels will be sorted by
+	// the value of key2 and put behind those pods have key1.
+	OrderPriority []UpdatePriorityOrderTerm `json:"orderPriority,omitempty"`
+	// Weight priority terms, pods will be sorted by the sum of all terms weight.
+	WeightPriority []UpdatePriorityWeightTerm `json:"weightPriority,omitempty"`
+}
+
+// UpdatePriorityOrder defines order priority.
+type UpdatePriorityOrderTerm struct {
+	// Calculate priority by value of this key.
+	OrderedKey string `json:"orderedKey"`
+	// Type of value to be compared and sorted.
+	// Values will be sorted from biggest to smallest by default.
+	// Defaults to String.
+	ValueType UpdatePriorityOrderValueType `json:"valueType,omitempty"`
+}
+
+type UpdatePriorityOrderValueType string
```

Comment:
How about renaming it to OrderType:
1. numerical
2. alphabetical 
3. version (https://github.com/ekg/filevercmp)